### PR TITLE
Add API functions

### PR DESCRIPTION
**Pull request by Shreyas-S-Raman**
Added api_support_functions.py file:

- dreamerv2_encoder()
- dreamerv2_reconstruction()
- dreamerv2_error()

### CODE REVIEW

File: dreamerv2/api_support_functions.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        self.agnt.load(self.logdir / 'saved_models' / model_file)
+
+        self.policy = common.RandomAgent(act_space)
+        self.state = None
+
+        self.obs_keys = {'reward':0.0,'is_last':False,'is_terminal':False,'is_timeout':False,'is_first':True}
+        
+
+    def dreamerv2_encoder(self, obs):
+        # TODO: change observation_dict input to observation
```
Comment (camall3n) [inline dreamerv2/api_support_functions.py]: Remove TODO comments

File: dreamerv2/api_support_functions.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        initialize_agent(next(train_dataset))
+        self.agnt.load(self.logdir / 'saved_models' / model_file)
+
+        self.policy = common.RandomAgent(act_space)
+        self.state = None
+
+        self.obs_keys = {'reward':0.0,'is_last':False,'is_terminal':False,'is_timeout':False,'is_first':True}
+        
+
+    def dreamerv2_encoder(self, obs):
```
Comment (camall3n) [inline dreamerv2/api_support_functions.py]: Change name to `encode`

File: dreamerv2/api_support_functions.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+        # TODO: check if we can do length-1 trajectories (Priority)
+        states, _ = self.agnt.wm.rssm.observe(embed[:1, :1], observation_dict['action'][:1, :1],
+                                              observation_dict['is_first'][:1, :1])
+
+        features = self.agnt.wm.rssm.get_feat(states)
+
+        return features
+
+    def dreamerv2_reconstruction(self, features):
```
Comment (camall3n) [inline dreamerv2/api_support_functions.py]: Change name to `decode`

File: dreamerv2/api_support_functions.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        # TODO: reconstruction uses features only, no observation_dict
```
Comment (camall3n) [inline dreamerv2/api_support_functions.py]: Remove TODO comments

**Review by camall3n — CHANGES_REQUESTED** [review]

File: dreamerv2/evaluate_reconstructions.py [hunk 5]
```diff
@@ -24,113 +30,56 @@
...
-
     args = parser.parse_args()
 
     logdir = pathlib.Path(args.model_path).expanduser()
 
     #check if logdir is valid
-    if not logdir.exists() or not (logdir/'saved_models').exists() or os.listdir((logdir/'saved_models'))==0:
+    if not logdir.exists() or not (logdir / 'saved_models').exists() or os.listdir(
+        (logdir / 'saved_models')) == 0:
         raise RuntimeError(f'Models not found in {logdir} or [...]/saved_models')
```
Comment (camall3n) [inline dreamerv2/evaluate_reconstructions.py:72]: Copy this check over to api_support_functions.py in place of the `NotImplementedError`

File: dreamerv2/api_support_functions.py (hunk 1 above)
Comment (Shreyas-S-Raman) [reply dreamerv2/api_support_functions.py]: Done

File: dreamerv2/api_support_functions.py (hunk 2 above)
Comment (Shreyas-S-Raman) [reply dreamerv2/api_support_functions.py]: Done

File: dreamerv2/api_support_functions.py (hunk 3 above)
Comment (Shreyas-S-Raman) [reply dreamerv2/api_support_functions.py]: Done

File: dreamerv2/api_support_functions.py (hunk 4 above)
Comment (Shreyas-S-Raman) [reply dreamerv2/api_support_functions.py]: Done

File: dreamerv2/evaluate_reconstructions.py (hunk 5 above)
Comment (Shreyas-S-Raman) [reply dreamerv2/evaluate_reconstructions.py:72]: Done

**Review by camall3n — APPROVED** [review]

**Review by camall3n — CHANGES_REQUESTED** [review]
Please ensure this branch is up to date with the most recent commit on `factored`.

```bash
git checkout updated-logging
git fetch
git merge origin/factored

# fix any issues if necessary & complete the merge

git push origin updated-logging
```

**Review by camall3n — APPROVED** [review]

**Pull request merged**